extensions/hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3Repository.java

```diff
@@ -162,26 +169,34 @@ private ObjectMetadata createObjectMetadata(final String mdMD5Hash16, final Stri
     }
 
     @Override
-    public void deleteBySha1(final String sha1Hash) {
-        LOG.info("Deleting S3 object from bucket {} and key {}", s3Properties.getBucketName(), sha1Hash);
-        amazonS3.deleteObject(new DeleteObjectRequest(s3Properties.getBucketName(), sha1Hash));
+    public void deleteBySha1(final String tenant, final String sha1Hash) {
+        final String key = objectKey(tenant, sha1Hash);
+
+        LOG.info("Deleting S3 object from bucket {} and key {}", s3Properties.getBucketName(), key);
+        amazonS3.deleteObject(new DeleteObjectRequest(s3Properties.getBucketName(), key));
+    }
+
+    private static String objectKey(final String tenant, final String sha1Hash) {
+        return tenant.trim().toUpperCase() + "/" + sha1Hash;
```

Comment:
small one: you could call the `sanitizeTenant()` method which does exactly this `tenant.trim().toUpperCase()`